bar_option: Add BarFillerClearOnAbort helper function

mustafabayar opened the pull request:
We currently have helper function to clear the progress bar on complete but not on abort. I recently wanted to use this library and I need to clear the bar on both events. Although it can be achieved using `BarFillerMiddleware` directly, it would be nice to have this shortcut helper function.
Should we also add **CompleteOrAbort** version? If we add both `BarFillerClearOnAbort` and `BarFillerClearOnComplete` options to the bar, would it cause problems as both calls `base.Fill(w, st)` ?

mustafabayar committed:
bar_option: Add BarFillerClearOnAbort helper function

We currently have helper function to clear the progress bar on complete
but not on abort. Although it can be achieved using BarFillerMiddleware
directly, it would be nice to have this helper function.

vbauerster commented:
Ok, can you please add empty `Copyright Waiver` commit as described in [CONTRIBUTING](https://github.com/vbauerster/mpb/blob/master/CONTRIBUTING).

> Should we also add CompleteOrAbort version? If we add both BarFillerClearOnAbort and BarFillerClearOnComplete options to the bar, would it cause problems as both calls base.Fill(w, st) ?

No, because opt in both is CompleteOrAbort in essence and it won't cause any problems. It is either on complete or on abort but never both, abort is no-op after complete.

mustafabayar committed:
mustafabayar Copyright Waiver

I dedicate any and all copyright interest in this software to the
public domain.  I make this dedication for the benefit of the public at
large and to the detriment of my heirs and successors.  I intend this
dedication to be an overt act of relinquishment in perpetuity of all
present and future rights to this software under copyright law.

mustafabayar commented:
> Ok, can you please add empty `Copyright Waiver` commit as described in [CONTRIBUTING](https://github.com/vbauerster/mpb/blob/master/CONTRIBUTING).
> 
> > Should we also add CompleteOrAbort version? If we add both BarFillerClearOnAbort and BarFillerClearOnComplete options to the bar, would it cause problems as both calls base.Fill(w, st) ?
> 
> No, because opt in both is CompleteOrAbort in essence and it won't cause any problems. It is either on complete or on abort but never both, abort is no-op after complete.

Done!
